crates/pico/tests/try_ok.rs

```diff
@@ -0,0 +1,125 @@
+use pico::{Database, MemoRef, SourceId, Storage};
+use pico_macros::{Db, Source, memo};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use thiserror::Error;
+
+static TRACK_CLONE_COUNT: AtomicUsize = AtomicUsize::new(0);
+
+#[derive(Db, Default)]
+struct TestDatabase {
+    storage: Storage<Self>,
+}
+
+#[test]
+fn try_ok_projects_ok_value() {
+    let mut db = TestDatabase::default();
+
+    let id = db.set(Input {
+        key: "key",
+        value: "asdf".to_string(),
+    });
+
+    assert_eq!(*process_input(&db, id).unwrap(), 'a');
+}
+
+#[test]
+fn projected_memo_correctly_restored() {
+    let mut db = TestDatabase::default();
+
+    let id = db.set(Input {
+        key: "key",
+        value: "asdf".to_string(),
+    });
+
+    let result = process_input(&db, id).unwrap();
+    assert_eq!(*consume_result(&db, result), 'a');
```

Comment:
What are we testing in this test that we aren't testing in `try_ok_projects_ok_value`?

Reply:
we check if memo_ref that were passed as an argument is correctly cloned from the store with proper projector